### PR TITLE
Implement Scan/Verify as functions

### PR DESCRIPTION
**Pull request by czgdp1807**
Closes https://github.com/lfortran/lfortran/issues/1067

### CODE REVIEW

**Comment by czgdp1807** [comment]
For implementing ``scan`` and ``verify`` we need support for ``present`` intrinsic because ``back`` is an optional argument.

**Comment by certik** [comment]
Yes, we have to implement `present` anyway.

File: src/lfortran/semantics/ast_common_visitor.h [hunk 1]
```diff
@@ -2717,10 +2717,53 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
...
+        std::string function_name = func_name + "_kind" + std::to_string(kind_value);
+
+        ASR::call_arg_t string_arg, set_arg, back_arg;
+        string_arg.loc = string->base.loc;
+        string_arg.m_value = string;
+        if( set ) {
+            set_arg.loc = set->base.loc;
+        } else {
+            set_arg.loc = x.base.base.loc;
+        }
```
Comment (Thirumalai-Shaktivel) [inline src/lfortran/semantics/ast_common_visitor.h:2740]: `set` will always be there, right?
`RESULT = SCAN(STRING, SET[, BACK [, KIND]])`

**Comment by Thirumalai-Shaktivel** [comment]
LGTM

**Review by certik — APPROVED** [review]
I think this is good to go in, after conflicts are resolved.

**Pull request merged**